sources/packages/backend/apps/api/src/services/application-change-request/application-change-request.service.ts

```diff
@@ -0,0 +1,229 @@
+          );
+        }
+        // End the workflow.
+        await this.workflowClientService.sendApplicationChangeRequestStatusMessage(
+          applicationId,
+          applicationEditStatus,
+        );
+        return;
+      }
+      const {
+        previousCompletedApplicationId,
+        previousCompletedApplicationCurrentAssessment,
+        newApplicationCurrentAssessment,
+      } = await this.getPreviousCompletedApplication(
+        applicationId,
+        transactionalEntityManager,
+      );
+      // Copy the most recent offering id and the student appeal id
+      // from the latest preceding application to the newly approved current application.
+      newApplicationCurrentAssessment.offering = {
+        id: previousCompletedApplicationCurrentAssessment.offering?.id,
+      } as EducationProgramOffering;
+      if (previousCompletedApplicationCurrentAssessment.studentAppeal) {
+        newApplicationCurrentAssessment.studentAppeal = {
+          id: previousCompletedApplicationCurrentAssessment.studentAppeal.id,
+        } as StudentAppeal;
+      }
+      // Update the previously completed application to be in Edited status.
+      await transactionalEntityManager.getRepository(Application).update(
+        {
+          id: previousCompletedApplicationId,
+        },
+        {
+          applicationStatus: ApplicationStatus.Edited,
+          modifier: auditUser,
+          updatedAt: currentDate,
+        },
+      );
+      const updateResult = await transactionalEntityManager
+        .getRepository(Application)
+        .update(
+          {
+            id: applicationId,
+            applicationEditStatus: ApplicationEditStatus.ChangePendingApproval,
+          },
+          {
+            applicationEditStatus: ApplicationEditStatus.ChangedWithApproval,
+            applicationStatus: ApplicationStatus.Completed,
+            currentAssessment: newApplicationCurrentAssessment,
+            modifier: auditUser,
+            updatedAt: currentDate,
+          },
+        );
+      if (!updateResult.affected) {
+        throw new Error(
+          `Application ${applicationId} to assess change not found or not in valid status to be updated.`,
+        );
+      }
+      // Send a message to the workflow to proceed.
+      await this.workflowClientService.sendApplicationChangeRequestStatusMessage(
+        applicationId,
+        applicationEditStatus,
+      );
+    });
+  }
+
+  /**
+   * Get the most recent application with the same parent that has a
+   * Change with approval or Original application edit status.
+   * @param applicationId application id.
+   * @param transactionalEntityManager transactional entity manager.
+   * @returns the most recent application details with the same parent
+   * that has a Change with approval or Original application edit status.
+   */
+  private async getPreviousCompletedApplication(
```

Comment:
This method may be removed if the suggested query is used or this can be simplified with the proposed query.
https://github.com/bcgov/SIMS/pull/4693/files#r2089938053